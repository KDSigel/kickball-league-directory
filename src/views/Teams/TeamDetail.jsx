import React, { useEffect, useState } from 'react'
import { Link, useHistory, useParams } from 'react-router-dom'
import { deletePlayerById } from '../../services/players'
import { deleteTeamById, getTeamById } from '../../services/teams'

export default function TeamDetail() {

    const [team, setTeam] = useState({})
    const [loading, setLoading] = useState(true)
    const {id} = useParams()
    const history = useHistory()

    useEffect(() => {
        async function getTeamDeets() {
            const response = await getTeamById(id)
            setTeam(response)
            setLoading(false)
        }
        getTeamDeets()
    }, [id]
    )

    const handleEdit = () => {
    history.push(`/team/edit/${id}`)
    }

    const handleDelete = async () => {
        await team.players.map((player) => deletePlayerById(player.id))
        await deleteTeamById(id);
        await history.push('/teams/')
    }

    return (
        <div>
            {loading ? <h1>'loading'</h1>
                    : <>
                    <h2>{team.name}</h2>
                    <p>{team.city}, {team.state}</p>
                    <ul>
                        {team.players.map((player) => {
                            return (
                            <li key={player.id}>
                                <Link to={`/player/${player.id}`}>
                                {player.position}: {player.name}
                                </Link>
                            </li>
                            );
                        })}
                    </ul>
                    <button 
                        type='button' 
                        onClick={handleEdit}
                        >
                            Edit team
                    </button>
                    <button
                        type="button"
                        aria-label='delete team'
                        onClick={handleDelete}
                        >
                            Delete
                    </button>
                    </>
            }
        </div>
    )
}
