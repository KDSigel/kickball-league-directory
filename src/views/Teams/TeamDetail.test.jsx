import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import TeamDetail from './TeamDetail';

const mockTeam1 = {
    id: 1,
    created_at: '2021-12-08T20:26:24.408898+00:00',
    name: 'Hot Stuff',
    city: 'Kirkland',
    state: 'Wa',
    players: [],
  };
  
  const server = setupServer(
    rest.get('https://byaeuvaytqhkrzznhwia.supabase.co/rest/v1/teams', (req, res, ctx) => {
      return res(ctx.json(mockTeam1))
    })
  );
  
  beforeAll(() => {
    server.listen();
  });
  
  afterAll(() => {
    server.close();
  });

it('it should check TeamDetail', async () => {

    const {container} = render(
        <MemoryRouter initialEntries={['/teams/1']}>
            <Switch>
                <Route exact path='/teams/:id' component={TeamDetail}/>
            </Switch>
        </MemoryRouter>
    )

    screen.getByText('loading', {exact: false})
    const oneTeamName = await screen.findByText('Edit team', {exact: false})
    expect(oneTeamName).toBeInTheDocument()

    expect(container).toMatchSnapshot()

}
)

// tests by Karl