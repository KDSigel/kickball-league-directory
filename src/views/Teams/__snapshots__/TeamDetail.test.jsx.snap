// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should check TeamDetail 1`] = `
<div>
  <div>
    <h2>
      Hot Stuff
    </h2>
    <p>
      Kirkland
      , 
      Wa
    </p>
    <ul />
    <button
      type="button"
    >
      Edit team
    </button>
    <button
      aria-label="delete team"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`it should check team details 1`] = `
<div>
  <div>
    <h2>
      Hot Stuff
    </h2>
    <p>
      Kirkland
      , 
      Wa
    </p>
    <ul />
    <button
      type="button"
    >
      Edit team
    </button>
    <button
      aria-label="delete team"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
